services/docker/docker-helpers.js

```diff
@@ -1,16 +1,66 @@
 'use strict'
 
 const dockerBlue = '066da5' // see https://github.com/badges/shields/pull/1690
+const Joi = require('@hapi/joi')
+const { latest } = require('../version')
 
-function buildDockerUrl(badgeName) {
-  return {
-    base: `docker/${badgeName}`,
-    pattern: ':user/:repo',
+const queryParamSchema = Joi.object({
+  sort: Joi.string()
+    .valid('date', 'semver')
+    .default('date'),
+}).required()
+
+function buildDockerUrl(badgeName, includeTagRoute, includeQueryParam) {
```

Comment:
Feels like we've got too much service-specific stuff in here now. Let's extract things from here that are not generically common (like the queryParamSchema for sort order for Versions) into the service that's actually using it.